### PR TITLE
added check for shesha and compass versions

### PR DESCRIPTION
**Pull request by HorstSteuer**
A proposal to solve problems as stated in e.g., #17 . When shesha and compass versions are not compatible, the upcoming error messages are not very helpful.
This proposed solution simply checks that the binary COMPASS and shesha are of the same version. This makes the assumption that the versions have to match exactly, which might be wrong, though?

### CODE REVIEW

**Comment by a-sevin** [comment]
Thanks Horst! It was in my todo-list and I never implented it.

I check it very and I merge it

**Comment by HorstSteuer** [comment]
Added a bunch of fix proposals. Most notable are probably the call to apply_control in GenericSupervisor, which was not done for all controllers due to an erroneous indentation. Similar problem for generate_pseudo_source() in util.py.

File: shesha/util/psfMap.py [hunk 1]
```diff
@@ -140,11 +138,11 @@ def setLGS(self, LGS, LGSy=None):
...
     def setTS(self, TS, TSy=None):
         if (TSy is None):
-            self.TSx = LGS[0]
-            self.TSy = LGS[1]
+            self.TSx = TS[0]
+            self.TSy = TS[1]
         else:
-            self.TS = LGS
-            self.TSy = LGS
+            self.TS = TS
```
Comment (a-sevin) [inline shesha/util/psfMap.py:144]: I think it's `self.TSx = TS` I will ask to the guy who codes it ;)

File: shesha/scripts/closed_loop.py [hunk 2]
```diff
@@ -56,10 +56,23 @@
...
+    compass_version = subprocess.check_output('conda list | grep compass',shell=True).decode(
+            sys.stdout.encoding).split()[1]
+    assert(shesha_version == compass_version), 'SHESHA and COMPASS versions are not matching : %r != %r ' %\
+                                              (shesha_version, compass_version)
+
 
 if __name__ == "__main__":
     arguments = docopt(__doc__)
 
+    check_shesha_compass_versions()
```
[the remark this answers is not in the document]
Comment (HorstSteuer) [reply shesha/scripts/closed_loop.py:75]: Sounds good.

File: shesha/supervisor/compassSupervisor.py [hunk 3]
```diff
@@ -348,10 +347,7 @@ def export_config(self):
         Extract and convert compass supervisor configuration parameters
         into 2 dictionnaries containing relevant AO parameters
 
-        Args:
-            root: (object), COMPASS supervisor object to be parsed
```
[the remark this answers is not in the document]
Comment (HorstSteuer) [reply shesha/supervisor/compassSupervisor.py:352]: Yes, that is what I meant. I believe the argument 'root' is a leftover when this method was part of aoSupervisor in version 4.x  `def writeConfigOnFile(self, root=None)`
Now, the argument does not exist any more and can be deleted from the doc-string, I think.

**Comment by a-sevin** [comment]
I will work a bit on it. need to handle virtual env : 

To reproduce it, `conda env create .` then ` conda activate compass`  

Error:

```python
ipython -i shesha/scripts/closed_loop.py data/par/par4bench/scao_sh_16x16_8pix.py
Python 3.8.5 (default, Sep  4 2020, 07:30:14) 
Type 'copyright', 'credits' or 'license' for more information
IPython 7.14.0 -- An enhanced Interactive Python. Type '?' for help.
---------------------------------------------------------------------------
AssertionError                            Traceback (most recent call last)
~/workspace/anr-compass/shesha/shesha/scripts/closed_loop.py in <module>
     73     arguments = docopt(__doc__)
     74 
---> 75     check_shesha_compass_versions()
     76     param_file = arguments["<parameters_filename>"]
     77     compute_tar_psf = not arguments["--fast"]

~/workspace/anr-compass/shesha/shesha/scripts/closed_loop.py in check_shesha_compass_versions()
     66     compass_version = subprocess.check_output('conda list | grep compass',shell=True).decode(
     67             sys.stdout.encoding).split()[1]
---> 68     assert(shesha_version == compass_version), 'SHESHA and COMPASS versions are not matching : %r != %r ' %\
     69                                               (shesha_version, compass_version)
     70 

AssertionError: SHESHA and COMPASS versions are not matching : '5.0.0' != 'packages' 

In [1]: subprocess.check_output('conda list | grep compass',shell=True)                                                                                 
Out[1]: b'# packages in environment at /home/sevin/Packages/miniconda3/envs/compass:\ncompass                   5.0.0             py3.8_cuda110    compass\n'
```

**Comment by a-sevin** [comment]
I don't know how to modify your PR, I forked your fork and I did a PR on it :sweat_smile: ...
Waiting your merge before merging it

**Pull request merged**